src/pip/_internal/resolution/resolvelib/resolver.py

```diff
@@ -114,42 +115,21 @@ def resolve(self, root_reqs, check_supported_wheels):
 
     def get_installation_order(self, req_set):
         # type: (RequirementSet) -> List[InstallRequirement]
-        """Create a list that orders given requirements for installation.
+        """Get order for installation of requirements in RequirementSet.
 
-        The returned list should contain all requirements in ``req_set``,
-        so the caller can loop through it and have a requirement installed
-        before the requiring thing.
+        The returned list contains a requirement before another that depends on
+        it. This helps ensure that the environment is kept consistent as they
+        get installed one-by-one.
 
-        The current implementation walks the resolved dependency graph, and
-        make sure every node has a greater "weight" than all its parents.
+        The current implementation creates a topological ordering of the
+        dependency graph, while breaking any cycles in the graph at arbitrary
+        points. We make no guarantees are made about where the cycle would be
+        broken, other than they would be broken.
```

Comment:
Nit: `We make no guarantees are made about where the cycle would be broken` could be changed to either of

1. `We make no guarantees about where the cycle would be broken` 
2. `No guarantees are made about where the cycle would be broken`

Reply:
Fixed locally, will push a fix with the other code.